File: parseFields.py
# helper function to parse command line fields argument

def parseFields(fieldsStr):
	fields = []
	parsed = fieldsStr.split(",")
	for part in parsed:
		subParsed = part.split("-")
		if len(subParsed)==1:
			fields.append(int(part))
		elif len(subParsed)==2:
			fields+=range(int(subParsed[0]),int(subParsed[1])+1)
		
	return [field-1 for field in fields]

File: parseFields.R
parseFields <- function(fieldsStr) {
	fields <- integer(0)
	parsed = stringr::str_split(fieldsStr,",")[[1]]
	for(part in parsed) {
		subParsed <- stringr::str_split(part,"-")[[1]]
		if(length(subParsed) == 1) {
			fields <- c(fields, as.integer(part))
		} else if(length(subParsed) == 2) {
			fields <- c(fields, as.integer(subParsed[1]:subParsed[2]))
		}
	}

	return(fields)
}
